teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,397 @@
+
+Example:
+
+```
+artifacthub://tekton-catalog-tasks/task/git-clone@0.6
+```
+
+Using the above example, a user can do the following when referencing the Task:
+
+```yaml
+taskRef:
+  resolver: hub
+  name: "artifacthub://tekton-catalog-tasks/task/git-clone@0.6"
+```
+
+##### Cluster Resolver
+Based on `cluster` resolver's [params](https://tekton.dev/docs/pipelines/cluster-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+cluster://<namespace>/<kind>/<name>
+```
+
+**Note** that we use a scheme called `cluster` which is only meaningful to the cluster resolver.
+
+Example:
+
+```
+cluster://default/task/git-clone
+```
+
+Using the above example, a user can do the following when referencing the Task:
+
+```yaml
+taskRef:
+  resolver: "cluster"
+  name: "cluster://default/task/git-clone" 
+```
+
+#### Custom Resolvers
+When implementing a new custom resolver, just like users have the freedom to choose their parameters, they also have the freedom to define their `name` (or `url`) based on:
+
+```
+<scheme>://<location>[@<version>][#<selector>][?<params>]
+```
+
+The custom resolver must determine the schemes it wants to support and subscribe to (in case of stage 2). For each scheme, it must define the remainder of `location, version, selectors and params` as needed. An example `scheme` could be `<custom-resolver-name>`. A `location` may be made up of concrete units e.g. `location = <foo>/<bar>` (i.e. expect two parts) that are up to the discretion of the custom resolvers.
+
+The custom resolver must validate the `name` which will be passed within the `Remote Resolution CRD's` spec. The `name` (or `url`) can then be parsed based on the format for that `scheme` and used to fetch the remote resource.
+
+### Stage 2
+
+In this stage, we enable the Tekton controller to be able to identify the resolvers on the clusters that have subscribed to the scheme of the incoming [resolver reference](#resolver-reference).
+
+#### Prioritized Subscription to `scheme`s
+This mapping of `scheme` to `resolver` could be easily maintained in a `configMap` (or a [CRD](#using-crd-to-store-prioritized-subscription-configuration)) for the cluster. For [built-in resolvers](#built-in-resolvers), Tekton installation will be pre-configured with the schemes and priorities so that the cluster operators do not need to worry about any interaction unless they want to update the priority or additional custom resolvers. The cluster operators will only be required to know the custom remote resolvers that are running on the cluster and the schemes they are subscribed to. They can modify the existing configuration to add the custom resolvers. This means that any resolver that is not a [built-in](#built-in-resolvers) resolver needs to be explicitly added to this configMap. The configMap must list out all the schemes supported by all the resolvers in order of priority.
```

Comment:
This will need to be highlighted when we implement this, because it is a change from what happens today (aka, today you install the resolver and you can use it directly).

Reply:
Agreed!